framework/src/mesh/MooseMesh.C

```diff
@@ -1281,6 +1288,10 @@ MooseMesh::detectPairedSidesets()
   {
     Elem * elem = *el;
 
+    // dimension of the current element and its normals
+    unsigned int i = elem->dim() - 1;
```

Comment:
Might as well also change the loops above and below to match the variable name...